File: solutions/0206/1.java
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode prev = null;
        ListNode curr = head;
        while (curr != null) {
            ListNode cnext = curr.next;
            if (prev == null) {
                curr.next = null;
            } else {
                curr.next = prev;
            }
            prev = curr;
            curr = cnext;
        }
        return prev;
    }
}

File: solutions/0206/1.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* curr = head;
        ListNode* prev = nullptr;

        while (curr != nullptr) {
            ListNode* cnext = curr->next;
            if (prev == nullptr) {
                curr->next = nullptr;
            } else {
                curr->next = prev;
            }

            prev = curr;
            curr = cnext;
        }
        
        return prev;
    }
};
